### PR TITLE
feat(slack): Add ability to send workflow notification to thread

### PR DESCRIPTION
**Pull request by ykamo001**
- Add logic to handle sending an activity's data to a slack thread
  - Add tests
- Create new class and helper functions to consolidate repeated logic
  - Add logging for all places where things can go wrong
  - Breakup functions for more modularity and singular responsibilities 
- Invoke new class and logic in the proper location
  - Guard logic behind a FF

Requires: https://github.com/getsentry/sentry/pull/68142
Resolves: https://github.com/getsentry/team-core-product-foundations/issues/188

### CODE REVIEW

File: src/sentry/integrations/slack/service.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        if activity.group is None:
+            self._logger.info(
+                "no group associated on the activity, nothing to do",
+                extra={
+                    "activity_id": activity.id,
+                },
+            )
+            return None
+
+        if activity.user_id is None:
```
Comment (cathteng) [inline src/sentry/integrations/slack/service.py:60]: will we support system activities in the future?

File: src/sentry/integrations/slack/service.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                f"failed to find rule action {parent_notification.rule_action_uuid} for rule {rule.id}"
+            )
+
+        channel_id: str | None = rule_action.get("channel_id", None)
+        if not channel_id:
+            raise RuleDataError(
+                f"failed to get channel_id for rule {rule.id} and rule action {parent_notification.rule_action_uuid}"
+            )
+
+        block = self._slack_block_builder.get_markdown_block(text=notification_to_send)
```
Comment (cathteng) [inline src/sentry/integrations/slack/service.py:166]: so we are sending a markdown block of the context

File: src/sentry/integrations/slack/service.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        payload = {"channel": channel_id, "thread_ts": parent_notification.message_identifier}
+        slack_payload = self._slack_block_builder._build_blocks(
+            block, fallback_text=notification_to_send
+        )
+        payload.update(slack_payload)
+        client.post("/chat.postMessage", data=payload, timeout=5)
```
Comment (cathteng) [inline src/sentry/integrations/slack/service.py:172]: should we be try/excepting this and logging stuff if it fails?

Comment (cathteng) [reply src/sentry/integrations/slack/service.py:172]: also this function feels a little fat. if we're going to use this service for other things maybe we could split some of the stuff out (later)?

File: src/sentry/integrations/slack/service.py (hunk 1 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:60]: I think this is going to be a product decision, there's no technical limitations, other than maybe increasing our load and frequency on notifications

File: src/sentry/integrations/slack/service.py (hunk 2 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:166]: yeah we just need a 1 line sentence

File: src/sentry/integrations/slack/service.py (hunk 3 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:172]: we're catching the exception at the for loop level, but I guess it wouldn't hurt to output the payload so we have more visibility, let me add that

Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:172]: this particular function, like `_handle_parent_notification`?
I think these functions feel fat because of the logging, but I can try to consolidate a little more. Just don't want to preemptively over engineer anything

File: src/sentry/integrations/slack/service.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+from sentry.services.hybrid_cloud.integration import integration_service
+
+_default_logger = getLogger(__name__)
+
+
+class RuleDataError(Exception):
+    pass
+
+
+class SlackService:
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/service.py:24]: We have a lot of classes that look like they're the main slack handler 😅 Can you please add some documentation on what this handles.

File: src/sentry/integrations/slack/service.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        if not notification_to_send:
+            self._logger.info(
+                "notification to send is invalid",
+                extra={
+                    "activity_id": activity.id,
+                },
+            )
+            return None
+
+        try:
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/service.py:80]: We probably have this block of code everywhere we're working with slack. Can you make this as well as it's error logging to a util function for unified debugging experience.

Using it in other callsites can be a different PR.

File: src/sentry/integrations/slack/service.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+            integration = integration_service.get_integration(
+                organization_id=activity.project.organization_id,
```
Comment (sentaur-athena) [inline src/sentry/integrations/slack/service.py:82]: What I mean is that not finding integration doesn't have anything to do with activity but rather organization. So the code would look like this:

```
organization_id = activity.project.organization_id
# can organization be None? Check here instead of integration related code
integration = getIntegrationForOrganization() # this method has the try catch inside it
```

Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:82]: Yeah that makes sense! I can def add that helper function

File: src/sentry/integrations/slack/service.py (hunk 5 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:80]: Yeah that's a good call out

File: src/sentry/integrations/slack/service.py (hunk 4 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py:24]: Yeah fair, this should be the main business logic handler going forward, I'll add a doc string

File: src/sentry/integrations/slack/service.py (hunk 3 above)
Comment (ykamo001) [reply src/sentry/integrations/slack/service.py]: done, added more logging

**Review by sentaur-athena — APPROVED** [review]
